examples/calibration/plot_calibration_curve.py

```diff
@@ -5,130 +5,289 @@
+    index=index,
+)
+score_df.round(3)
+
+# %%
+# Next, we will compare:
+#
+# * :class:`~sklearn.linear_model.LogisticRegression` (baseline)
+# * :class:`~sklearn.svm.LinearSVC` (:term:`decision_function` scores
+#   min-max scaled to [0,1], since they are not proper probabilities)
+# * :class:`~sklearn.svm.LinearSVC` with isotonic and sigmoid
+#   calibration (see :ref:`User Guide <calibration>`)
+#
 
-def plot_calibration_curve(est, name, fig_index):
-    """Plot calibration curve for est w/o and with calibration. """
-    # Calibrated with isotonic calibration
-    isotonic = CalibratedClassifierCV(est, cv=2, method='isotonic')
 
-    # Calibrated with sigmoid calibration
-    sigmoid = CalibratedClassifierCV(est, cv=2, method='sigmoid')
+class NaivelyCalibratedLinearSVC(LinearSVC):
+    """LinearSVC with `predict_proba` method that naively scales
+    `decision_function` output for binary classification."""
 
-    # Logistic regression with no calibration as baseline
-    lr = LogisticRegression(C=1.)
+    def fit(self, X, y):
+        super().fit(X, y)
+        df = self.decision_function(X)
+        self.df_min_ = df.min()
+        self.df_max_ = df.max()
 
-    fig = plt.figure(fig_index, figsize=(10, 10))
-    ax1 = plt.subplot2grid((3, 1), (0, 0), rowspan=2)
-    ax2 = plt.subplot2grid((3, 1), (2, 0))
+    def predict_proba(self, X):
+        """Min-max scale output of `decision_function` to [0,1].
 
-    ax1.plot([0, 1], [0, 1], "k:", label="Perfectly calibrated")
-    for clf, name in [(lr, 'Logistic'),
-                      (est, name),
-                      (isotonic, name + ' + Isotonic'),
-                      (sigmoid, name + ' + Sigmoid')]:
-        clf.fit(X_train, y_train)
-        y_pred = clf.predict(X_test)
-        if hasattr(clf, "predict_proba"):
-            prob_pos = clf.predict_proba(X_test)[:, 1]
-        else:  # use decision function
-            prob_pos = clf.decision_function(X_test)
-            prob_pos = \
-                (prob_pos - prob_pos.min()) / (prob_pos.max() - prob_pos.min())
+        Parameters
+        ----------
+        X : ndarray of shape (n_samples, n_features)
+            The data.
 
-        clf_score = brier_score_loss(y_test, prob_pos, pos_label=y.max())
-        print("%s:" % name)
-        print("\tBrier: %1.3f" % (clf_score))
-        print("\tPrecision: %1.3f" % precision_score(y_test, y_pred))
-        print("\tRecall: %1.3f" % recall_score(y_test, y_pred))
-        print("\tF1: %1.3f\n" % f1_score(y_test, y_pred))
+        Returns
+        -------
+        proba : ndarray of shape (n_samples, 2)
+            Predicted probabilities of the 2 classes.
+        """
+        df = self.decision_function(X)
+        calibrated_df = (df - self.df_min_) / (self.df_max_ - self.df_min_)
+        proba_class1 = np.clip(calibrated_df, 0, 1)
+        proba_class0 = 1 - proba_class1
+        proba = np.c_[proba_class0, proba_class1]
+        return proba
```

Comment:
I did this because it makes `y_proba` from `NaivelyCalibratedLinearSVC` the same shape as `y_proba` from `LogisticRegression`, as later on we do:
```Python
    brier.append(brier_score_loss(y_test, y_proba[:, 1]))
    logloss.append(log_loss(y_test, y_proba[:, 1]))
```

Note that I do not do this in `plot_compare_calibration` (we just return `np.clip(calibrated_df, 0, 1)`) as we don't compute any score metrics there.

Alternatively I can add some logic when we calculate the score:
```Python
    if name == 'Logistic':
        brier.append(brier_score_loss(y_test, y_proba[:, 1]))
        logloss.append(log_loss(y_test, y_proba[:, 1]))
    else:
        brier.append(brier_score_loss(y_test, y_proba))
        logloss.append(log_loss(y_test, y_proba))  
```

Reply:
I think it's fine as it is.